codegen/runner/runner.go

```diff
@@ -62,62 +80,63 @@ func main() {
 	}
 
 	configRoot := filepath.Dir(*configFile)
-	config := zanzibar.NewStaticConfigOrDie([]*zanzibar.ConfigOption{
-		zanzibar.ConfigFilePath(*configFile),
-	}, nil)
+	configProvider, err := config.NewYAML(config.File(*configFile))
+	checkError(
+		err, fmt.Sprintf("can not read config %q", *configFile),
+	)
+
 
-	configRoot, err := filepath.Abs(configRoot)
+	codegenConfig := new(codegenOptions)
+	err = configProvider.Get("").Populate(&codegenConfig)
```

Comment:
Is there every going to be some codegen-specific key to separate things out?

Reply:
This looks odd. Is there no top-level key to read from? Will there be other modules reading from this same config library?